packages/core/error.ts

```diff
@@ -1,3 +1,33 @@
-export class RoutingJSError extends Error {}
-export class RoutingJSAPIError extends RoutingJSError {}
-export class RoutingJSClientError extends RoutingJSError {}
+export interface ErrorProps {
+    status_code: number | undefined
+    status: string | undefined
+}
+
+export interface CommonErrorProps extends ErrorProps {
+    error_code: number | undefined
+    error: string | undefined
+}
+
+export class RoutingJSAPIError<T extends ErrorProps> extends Error {
+    readonly properties: T
+
+    constructor(message: string, properties: T) {
+        super(message)
+        this.properties = properties
+    }
+
+    toJSON() {
+        return {
+            message: this.message,
+            properties: this.properties,
+        }
+    }
+}
+
+export const assertError = <T extends ErrorProps>(e: RoutingJSAPIError<T>) => {
```

Comment:
Good idea, maybe we should move this to a different file to separate actual library code from test code. Maybe have a folder `util` with a `test.ts` file?